bindings_ffi/src/lib.rs

```diff
@@ -128,6 +152,317 @@ impl FfiXmtpClient {
     }
 }
 
+#[uniffi::export(async_runtime = "tokio")]
+impl FfiXmtpClient {
+    pub async fn register_identity(&self) -> Result<(), GenericError> {
+        self.inner_client.register_identity().await?;
+
+        Ok(())
+    }
+}
+
+#[derive(uniffi::Enum)]
+pub enum FfiSortDirection {
+    Unspecified = 0,
+    Ascending = 1,
+    Descending = 2,
+}
+
+impl FfiSortDirection {
+    pub fn from_i32(val: i32) -> Self {
+        match val {
+            0 => FfiSortDirection::Unspecified,
+            1 => FfiSortDirection::Ascending,
+            2 => FfiSortDirection::Descending,
+            _ => panic!("Invalid sort direction"),
+        }
+    }
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiPagingInfo {
+    limit: u32,
+    cursor: Option<FfiCursor>,
+    direction: FfiSortDirection,
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiCursor {
+    pub digest: Vec<u8>,
+    pub sender_time_ns: u64,
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiV2QueryRequest {
```

Comment:
Can we move all these V2 API calls into a separate file? This one is getting pretty huge and messy.